src/go/rdctl/pkg/factoryreset/factory_reset_windows.go

```diff
@@ -39,6 +41,34 @@ var (
 	pEnumProcesses = pKernel32.NewProc("K32EnumProcesses")
 )
 
+// CheckProcessWindows - returns true if Rancher Desktop is still running, false if it isn't
+// along with an error condition if there's a problem detecting that.
+//
+// It does this by calling `tasklist`, the Windows answer to ps(1)
+
+func CheckProcessWindows() (bool, error) {
+	cmd := exec.Command("tasklist", "/NH", "/FI", "IMAGENAME eq Rancher Desktop.exe", "/FO", "CSV")
+	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: CREATE_NO_WINDOW}
+	allOutput, err := cmd.CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("Failed to run %q: %w", cmd, err)
+	}
+	r := csv.NewReader(strings.NewReader(string(allOutput)))
+	for {
+		record, err := r.Read()
+		if err != nil {
+			if err != io.EOF {
+				return false, fmt.Errorf("Failed to csv-read the output for tasklist: %w", err)
+			}
+			break
+		}
+		if record[0] == "Rancher Desktop.exe" {
```

Comment:
We should check if `record` has any entries before dereferencing.

Reply:
Not sure how that's going to happen, as the csv reader skips empty lines, but easy enough to do.